src/hazmat/primecount.rs

```diff
@@ -0,0 +1,288 @@
+/// | 2048 | ~2^2000                    | ~2^-37                 |
+/// | 4096 | ~2^4043                    | ~2^-41                 |
+///
+/// *(Relative error is calculated as the first omitted term divided by / Li_approx(x)`)*
+///
+/// ## Theoretical Error
+///
+/// Assuming RH, we can use Schoenfeld’s bound `|π(x) - Li(x)| < (1 / 8π) * √x * ln x`. This gives a very small
+/// theoretical error bound compared to the best known unconditional bounds (by Trudgian).
+///
+/// | Bits | Error Bound (abs) | Error Bound (rel) |
+/// | :--- | :---------------- | :-----------------|
+/// | 1024 | < 2^517           | < 2^-497          |
+/// | 2048 | < 2^1030          | < 2^-1007         |
+/// | 4096 | < 2^2055          | < 2^-2029         |
+///
+/// *(Relative error bound is calculated as `Schoenfeld’s Bound (abs) / Li_approx(x)`)*
+///
+/// ## Discussion
+/// Assuming RH, the dominant error term in estimating `π(x)` with `Li_approx(x)` comes from truncating the asymptotic
+/// expansion of Li(x). While this truncation error is extremely small in relative terms, it is large in absolute terms.
+/// Improving the Li(x) approximation to be the same order of magnitude as the Schoenfeld bound would require using
+/// hundreds of terms from the asymptotic series (or employing more complex methods like continued fractions for the
+/// remainder, which is beyond the scope of this simple approximation). In relative terms the relative error bound is
+/// approximately ~2^-33 (for 1024 bits) down to ~2^-41 (for 4096 bits). This corresponds to an extremely small
+/// percentage error (significantly less than 10^-8 %).
+///
+/// It should be noted that while Li(x) is generally smaller than π(x) for 'small' x, it is known that the sign of
+/// `π(x) - Li(x)` changes infinitely often. It has been proven that there must be a crossing below ~10^316 (~2^1051),
+/// which is well within the ranges used in this library. Thus, users should be aware that the estimate provided here
+/// can be both greater than and smaller than the actual value of π(x).
+///
+/// # Sources
+///
+/// Wikipedia, [Prime Counting Function][wikipedia-pcf]
+/// Pierre Dusart (2010), [ESTIMATES OF SOME FUNCTIONS OVER PRIMES WITHOUT R.H.][dusart2010].
+/// Pierre Dusart (2018), [Explicit estimates of some functions over primes][dusart2018].
+/// Lowell Schoenfeld (1976), [Sharper Bounds for the Chebyshev Functions θ(x) and ψ(x)](schoenfeld).
+/// Trudgian, T. S. (2014). [Updating the error term in the prime number theorem](trudgian)
+///
+/// [dusart2010]: https://arxiv.org/pdf/1002.0442
+/// [dusart2018]: https://www.researchgate.net/publication/309522478_Explicit_estimates_of_some_functions_over_primes
+/// [wikipedia-pcf]: https://en.wikipedia.org/wiki/Prime-counting_function
+/// [schoenfeld]: https://www.jstor.org/stable/2005976
+/// [trudgian]: https://arxiv.org/abs/1401.2689
+///
+pub fn estimate_pi_x<const LIMBS: usize, const RHS_LIMBS: usize>(x: &Uint<LIMBS>) -> Uint<LIMBS>
```

Comment:
Would this be better named something less mathematical, like `estimate_primecount`?